gravitee-node-vertx/src/main/java/io/gravitee/node/vertx/configuration/HttpServerConfiguration.java

```diff
@@ -592,7 +593,11 @@ private List<String> getArrayValues(String prefix, List<String> defaultValue) {
         }
 
         public HttpServerConfiguration build() {
-            Assert.notNull(environment, "Environment can not be null. Call withEnvironment method first to configured it");
+            Assert.notNull(environment, "Environment can not be null. Call withEnvironment() method first to configured it");
```

Comment:
With the new changes that David asked for, I don't think it will be backward compatible anyway (the initial changes did't break anything).
The idea is to create multiple HTTP servers by the factory bean instead of just one. (based on listener.http, listener.https ...)
Right now there is just HttpServerFactorty and configuration, later NetServerFactory and NetServerConfigurations can be also added for the tcp part

Reply:
Hum... not sure to understand why we can't keep the backward compatibility and continue to support `http.xxxx` if no `listeners.name.xxxx` is defined.
Anyway, I have several questions:

- How do you plan to deal with the keystone / truststore part as it is currently made to be a single global one. Here it seems to be segregated per listener so I'm wondering how it will work especially if we want to deal with MTLS plan for APIM.
- Related to the above question, if my understanding is correct, the listeners are named. For example  `listeners.http` and `listener.https`, **`http`** and **`https`** are just names and they could be **`unsecured`,** 
-  and **`secured`** instead, right? You mentioned it is related to `HttpServerFactory`, `NetServerFactory` but I don't see well how you make that link (there is no `type` or something).
- I don't see any check preventing the user to provide the same port twice, is it something you plan to add later?

Reply:
This part is not an issue Jeoffrey, I can update this part if you are happy with the rest of the code. Probably it would better if I created a draft PR actually.
- How do you plan to deal with the keystone ....  ==> Each server has its own settings and can have different settings, once secure, once with mTLS and so on. The question would be how we are going to make sure we are not serving a request from unsecured one while it should be handle by the mTLS listener. That I really don't know what possibilities we have at the moment in the API definition level.
- the listeners are named. Yes this is what I understood from our talk with David but the name really doesn't matter because you can use the prefix to parse something else like `unsecured`
- My proposal was not let the user decide about the security, so if it suppose to be http, then all the tls part should be disabled and the `secured` one should be by default tls and throws an error if it is not configured properly. David has other opinions it seems.
- there is no type or something: listeners.http(s) HttpServerFactory listeners.tcp(s) NetServerFactory
- I don't see any check preventing, good point, I'll update the code but at the end the server will fail to start in the runtime if that will be the case